src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -850,14 +873,23 @@ public String getRemoteUrl(String name) throws GitException {
      * @throws hudson.plugins.git.GitException if underlying git operation fails.
      */
     @NonNull
-    public Repository getRepository() throws GitException {
+    private Repository getRepository() throws GitException {
         try {
             return new RepositoryBuilder().setWorkTree(workspace).build();
         } catch (IOException e) {
             throw new GitException(e);
         }
     }
 
+    public <T> T withRepository(RepositoryCallback<T> callable) throws IOException, InterruptedException {
+        Repository repo = getRepository();
+        try {
```

Comment:
if repository is `AutoCloseable` then try-with-resources is better here - if both the callable and the close fails the initial error would be swallowed with the code as is :-(